frontend/src/queries/nodes/SavedInsight/SavedInsight.tsx

```diff
@@ -30,10 +31,7 @@ export function SavedInsight({ query, context, cachedResults }: InsightProps): J
         throw new Error('InsightNode expects an insight with filters')
     }
 
-    return (
-        <Query
-            query={{ ...query, kind: NodeKind.InsightVizNode, source: filtersToQueryNode(insight.filters) }}
-            context={{ ...context, insightProps }}
-        />
-    )
+    const query = { ...propsQuery, ...dataQuery, full: propsQuery.full }
+
+    return <Query query={query} context={{ ...context, insightProps }} />
```

Comment:
We should pass through `setQuery` from props, so that one can edit a saved insight inside notebooks.

Reply:
I did wonder that but actually left it out purposefully because we don't want to support editing of saved insights in notebooks.

Given the `NodeKind.SavedInsightNode` gets transformed into a `DataTableNode` or `InsightVizNode` I assumed that the `setQuery` would then only edit the generated query and not the saved insight (which I believe gets updated internally somehow).

We basically don't need to update the generated query for the notebooks use case but can add it if it makes sense more generally

Reply:
Ah ok, didn't know it's not needed for notebooks. Let's leave this as it is for now then.